src/reverselock.h

```diff
@@ -13,7 +13,7 @@ class reverse_lock
 {
 public:
 
-    explicit reverse_lock(Lock& lock) : lock(lock) {
+    explicit reverse_lock(Lock& _lock) : lock(_lock) {
```

Comment:
In this particular case, the requirement for same-binary makes the diff longer than needed. But I agree that in this case, it is better to have the same binary just to be sure.